src/backend/metal/src/device.rs

```diff
@@ -192,6 +192,13 @@ impl PhysicalDevice {
 
     pub(crate) fn new(device: metal::Device) -> Self {
         let private_caps = PrivateCapabilities {
+            msl_version: if device.supports_feature_set(MTLFeatureSet::macOS_GPUFamily1_v3) {
```

Comment:
It appears to be unrelated to the feature set if we read between the lines in the [Metal Shading Language Specification (6.4)](https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf):

* ios-metal1.0
 – support the unified graphics / compute language revision 1.0 programs for iOS 8.
* ios-metal1.1
 – support the unified graphics / compute language revision 1.1 programs for iOS 9.
* ios-metal1.2
 – support the unified graphics / compute language revision 1.2 programs for iOS 10.
* ios-metal2.0
 – support the unified graphics / compute language revision 2.0 programs for iOS 11.
* ios-metal2.1
 – support the unified graphics / compute language revision 2.1 programs for iOS 12.
* osx-metal1.1
 – support the unified graphics / compute language revision 1.1 programs for macOS 10.11.
* osx-metal1.2
 – support the unified graphics / compute language revision 1.2 programs for macOS 10.12.
* osx-metal2.0
 – support the unified graphics / compute language revision 2.0 programs for macOS 10.13.
* osx-metal2.1
 – support the unified graphics / compute language revision 2.0 programs for macOS 10.14.

Reply:
So basically, it's the OS version and not the feature set. hmm